master/reference/public-cloud/aws.md

```diff
@@ -0,0 +1,152 @@
+
+IPv6 BGP status
+No IPv6 peers found.
+```
+
+## Allowing Workload-to-Workload Traffic
+
+By default, each EC2 instance performs source/destination checks on inbound and outbound traffic.
+This means that the instance must be the source or destination of any traffic it
+sends or receives.
+
+Since Calico allocates a unique IP to each container, container traffic leaving
+and entering the EC2 instance will have an "unrecognized" source address
+and traffic will be dropped.
+
+This behavior can be disabled for inter-VPC workloads,
+and can be circumvented for cross-VPC workloads.
+
+#### Inter-VPC: Disable Src/Dest Checks
+
+See the [AWS guide on disabling src/dest checks for an EC2 instance](http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_NAT_Instance.html#EIP_Disable_SrcDestCheck).
+
+#### Cross-VPC: Enable IP-in-IP
+
+Unfortunately, cross-VPC src/dest checks cannot be disabled in AWS.
+
+To circumvent this limitation, each Calico host can encapsulate workload-to-workload
+traffic using IP-in-IP encapsulation. AWS networking fabric will recognize the host
+addresses for these packets and will not drop the traffic.
+
+IP-in-IP is enabled on a per-pool basis with the following command:
+
+```yaml
+cat << EOF | calicoctl apply -f -
+apiVersion: v1
+kind: ipPool
+metadata:
+  cidr: 10.1.0.0/16
+spec:
+  ipip:
+    enabled: true
+EOF
+```
+
+## Allowing Workload-to-WAN Traffic
+
+In a private cloud where operators maintain full control of the networking
+fabric, NATting takes place at the edge router.
+
+In AWS, operators do not have full control of the networking fabric. As such,
+they cannot configure edge routers to NAT calico workload traffic.
+
+Fortunately, calico can perform NAT on outbound traffic on each compute node by
+enabling `nat-outgoing` on each pool:
+
+```yaml
+cat << EOF | calicoctl apply -f -
+apiVersion: v1
+kind: ipPool
+metadata:
+  cidr: 10.1.0.0/16
+spec:
+  nat-outgoing: true
+EOF
+```
+
+## (Optional) Modifying MTU for Performance Boost
+
+Some [AWS instance types](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/network_mtu.html#jumbo_frame_instances)
+utilize a default [MTU](https://en.wikipedia.org/wiki/Maximum_transmission_unit)
+of 9001, which is larger than the standard 1500 used by most of the Internet.
+
+If you have a high traffic deployment and all of your host instances in your
+AWS Calico cluster are using these instance types, you may be able to improve
+performance for traffic between your Calico nodes. To do this, you need to
+modify the MTU of the veth interfaces of your Calico containers.
+
+> **WARNING**: You must run these commands on ALL of the hosts and containers in your
+> deployment. If not all of your instances are jumbo frame instances or if you
+> do not modify MTU on all containers and hosts, you may  experience unexpected
+> behavior and/or packet loss.
+
+#### Modify Calico interface MTU on hosts
+
+First, modify the MTU on the `cali######`
```

Comment:
Oh man these are really manual instructions.

Can't we say "use the CNI plugin's MTU parameter" to do this?